### ISSUE TITLE
NextJS 9 - Compression?

### ISSUE THREAD

**Issue by ShanonJackson** [zeit/next.js#8042]
# Question about Next.js

I'll preface this with i'm still learning NextJS i saw the changes in version 9 and realized that there's no reason not to convert now with typescript "first-class-citizen" support; so i just spent the last 45 hrs converting a smaller side project before converting some bigger projects at work just to see what problems i run into.

How do you get gzipping working on NextJS 9? i'm using the inbuilt server (pages/api) and therefore not using the custom server.

Lighthouse is telling me none of my assets are being gzipped therefore punishing me but there's no "end point" to attach the compression express middle-ware to for client JS.

TL;DR using new NextJS 9 inbuilt server but can't figure out of the life of me how to get asset compression working; putting it in custom server stops my /pages/api endpoints working.

**Comment by blainegarrett** [comment]
@ShanonJackson Did you find a solution? Could you post it here for folks googling the same question?

**Comment by x4080** [comment]
I have question related to this question too, since I wanted to use compression, I use custom server, and with this custom server, how to use the new api feature in next js 9, or if it cannot, then this issue is very important (also for caching too)

**Comment by Timer** [comment]
Hey @ShanonJackson!

Compression of assets is typically handled by a L7 proxy like nginx -- a `next start` server should have a proxy in front of it performing most of these duties (Gzip, caching, etc). 

Deploying to ZEIT Now configures all of these things out of the box for you.

**Comment by franklinjavier** [comment]
How to make all these middlewares work without custom server?
Has anyone workaround?

![image](https://user-images.githubusercontent.com/440220/61926134-c0936480-af45-11e9-9d5d-399535b92013.png)

**Comment by Timer** [comment]
FYI we're adding compression by default: https://github.com/zeit/next.js/pull/8066

**Comment by ShanonJackson** [comment]
TL;DR compress outside of your application layer, nginx or your hosting provider should have that responsibility for those googling this.

**Comment by x4080** [comment]
@ShanonJackson i agree, now i use caddy server for compression and reverse proxy

**Comment by jolic** [comment]
Could you please post some information about how did you get caddy working with NextJS? Thx.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
